packages/core/config/webpack.common.js

```diff
@@ -36,66 +36,87 @@ module.exports = (webpackEnv) => {
+        : '[path][name]__[local]--[hash:base64:5]'
+
+      const loaderConfig = {
+        loader: 'css-loader',
+        options: {
+          sourceMap: !isEnvProduction,
+          importLoaders,
+        },
+      }
+
+      // Enables CSS modules
+      if (withModules) {
+        loaderConfig.options.modules = { localIdentName }
+      }
+
+      return loaderConfig
+    }
+
+    const redwoodPaths = getPaths()
+    const hasPostCssConfig = existsSync(redwoodPaths.web.postcss)
+
+    // We only use the postcss-loader if there is a postcss config file
+    // at web/config/postcss.config.js
+    const postCssLoader = hasPostCssConfig
+      ? {
+          loader: 'postcss-loader',
+          options: {
+            config: {
+              path: redwoodPaths.web.postcss,
+            },
+          },
+        }
+      : null
+
+    const numImportLoadersForCSS = hasPostCssConfig ? 1 : 0
+    const numImportLoadersForSCSS = hasPostCssConfig ? 2 : 1
+
     return [
       {
         test: /\.module\.css$/,
         loader: [
-          isEnvProduction ? MiniCssExtractPlugin.loader : 'style-loader',
-          {
-            loader: 'css-loader',
-            options: {
-              modules: {
-                localIdentName,
-              },
-              sourceMap: !isEnvProduction,
-            },
-          },
-        ],
+          styleOrExtractLoader,
+          cssLoader(true, numImportLoadersForCSS),
+          postCssLoader,
+        ].filter(Boolean),
       },
       {
         test: /\.css$/,
         loader: [
-          isEnvProduction ? MiniCssExtractPlugin.loader : 'style-loader',
-          {
-            loader: 'css-loader',
-            options: {
-              sourceMap: !isEnvProduction,
-            },
-          },
-        ],
+          styleOrExtractLoader,
+          cssLoader(false, numImportLoadersForCSS),
+          postCssLoader,
+        ].filter(Boolean),
+        sideEffects: true,
       },
       {
         test: /\.module\.scss$/,
         loader: [
-          isEnvProduction ? MiniCssExtractPlugin.loader : 'style-loader',
-          {
-            loader: 'css-loader',
-            options: {
-              modules: {
-                localIdentName,
-              },
-              sourceMap: !isEnvProduction,
-            },
-          },
+          styleOrExtractLoader,
+          cssLoader(true, numImportLoadersForSCSS),
+          postCssLoader,
           'sass-loader',
-        ],
+        ].filter(Boolean),
       },
       {
         test: /\.scss$/,
         loader: [
-          isEnvProduction ? MiniCssExtractPlugin.loader : 'style-loader',
-          {
-            loader: 'css-loader',
-            options: {
-              sourceMap: !isEnvProduction,
-            },
-          },
+          styleOrExtractLoader,
+          cssLoader(false, numImportLoadersForSCSS),
+          postCssLoader,
           'sass-loader',
-        ],
+        ].filter(Boolean),
+        sideEffects: true,
```

Comment:
Interesting, what does this do?

Reply:
I mirrored what create-react-app did with scss/css files. I believe (but not certain) it's for tree-shaking to keep the styles that aren't directly imported in JS from being removed.

https://webpack.js.org/guides/tree-shaking/#mark-the-file-as-side-effect-free

Reply:
Super interesting! I didn't know that was possible for css.